### PR TITLE
label-pull-requests: accept YAML as alternative to JSON

### PR DESCRIPTION
**Pull request by FnControlOption**
[homebrew-cask triage workflow](https://github.com/Homebrew/homebrew-cask/blob/master/.github/workflows/triage.yml#L17-L39)

```yaml
- name: Label pull request
  uses: Homebrew/actions/label-pull-requests@master
  if: always()
  with:
    token: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
    yaml: true
    def: |
      - label: new cask
        status: added
        path: Casks/.+

      - label: marked for removal/rejection
        status: removed
        path: Casks/.+

      - label: documentation
        path: (.*\.md|\.github/ISSUE_TEMPLATE/.*\.yml)

      - label: bump-cask-pr
        pr_body_content: Created with `brew bump-cask-pr`
```

[homebrew-core triage workflow](https://github.com/Homebrew/homebrew-core/blob/master/.github/workflows/triage.yml#L18-L164)

<details>

```yaml
- name: Label pull request
  uses: Homebrew/actions/label-pull-requests@master
  if: always()
  with:
    token: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
    yaml: true
    def: |
      - label: workflows
        path: .github/workflows/.+

      - label: new formula
        status: added
        path: Formula/.+

      - label: marked for removal/rejection
        status: removed
        path: Formula/.+

      - label: bottle unneeded
        path: Formula/.+
        content: \n  (bottle :unneeded)\n

      - label: no ARM bottle
        path: Formula/.+
        content: '\n    sha256.* big_sur: +"[a-fA-F0-9]+"\n'
        missing_content: '\n    sha256.* arm64_big_sur: +"[a-fA-F0-9]+"\n'

      - label: no Linux bottle
        path: Formula/.+
        content: \n  bottle do\n
        missing_content:
          - '\n    sha256.* x86_64_linux: +"[a-fA-F0-9]+"\n'
          - '\n    sha256.* all: +"[a-fA-F0-9]+"\n'
          - depends_on :macos

      - label: formula deprecated
        path: Formula/.+
        content: \n  deprecate!.*\n

      - label: formula disabled
        path: Formula/.+
        content: \n  disable!.*\n

      - label: legacy
        path: Formula/.+@.+
        except:
          - Formula/openssl@1.1.rb
          - Formula/openssl@3.0.rb
          - Formula/python@3.9
          - Formula/python-tk@3.9.rb

      - label: missing license
        path: Formula/.+
        missing_content: \n  license .+\n

      - label: deprecated license
        path: Formula/.+
        content: license .*"(GPL|LGPL|AGPL|GFDL)-[0-9].[0-9][+]?".*

      - label: go
        path: Formula/.+
        content: depends_on "go(@[0-9.]+)?"

      - label: haskell
        path: Formula/.+
        content: depends_on "(ghc|haskell-stack)(@[0-9.]+)?"

      - label: java
        path: Formula/.+
        content: depends_on "openjdk(@[0-9.]+)?"

      - label: linux-only
        path: Formula/.+
        content: depends_on :linux

      - label: macos-only
        path: Formula/.+
        content: depends_on :macos

      - label: lua
        path: Formula/.+
        content: depends_on "(lua|luajit|luajit-openresty)(@[0-9.]+)?"

      - label: nodejs
        path: Formula/.+
        content: depends_on "node(@[0-9.]+)?"

      - label: ocaml
        path: Formula/.+
        content: depends_on "ocaml(@[0-9.]+)?"

      - label: perl
        path: Formula/.+
        content: (depends_on|uses_from_macos) "perl(@[0-9.]+)?"

      - label: php
        path: Formula/.+
        content: (depends_on|uses_from_macos) "php(@[0-9.]+)?"

      - label: python
        path: Formula/.+
        content: (depends_on|uses_from_macos) "python(@[0-9.]+)?"
        missing_content: (depends_on|uses_from_macos) "python(@[0-9.]+)?" => \[?:(build|test)

      - label: ruby
        path: Formula/.+
        content: (depends_on|uses_from_macos) "ruby(@[0-9.]+)?"

      - label: rust
        path: Formula/.+
        content: depends_on "rust(@[0-9.]+)?"

      - label: dotnet
        path: Formula/.+
        content: depends_on "dotnet(@[0-9.]+)?"

      - label: swift
        path: Formula/.+
        content: system "swift", "build"

      - label: CI-long-timeout
        path: Formula/(boost|qt).rb
        keep_if_no_match: true

      - label: CI-build-dependents-from-source
        path: Formula/(cabal-install|docbook-xsl|emscripten|erlang|ghc|go|ocaml|ocaml-findlib|ocaml-num|openjdk|rust).rb
        keep_if_no_match: true

      - label: CI-skip-recursive-dependents
        path: Formula/(gettext|openssl).rb
        keep_if_no_match: true

      - label: CI-linux-wheezy
        path: Formula/(zlib|binutils).rb
        keep_if_no_match: true

      - label: CI-linux-self-hosted
        path: Formula/(gdbm|go|openssl@1.1|python@3.9|rust|sqlite).rb
        keep_if_no_match: true

      - label: bump-formula-pr
        pr_body_content: Created with `brew bump-formula-pr`
```

</details>

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
This looks really nice to me but I'm interested what @dawidd6 and other think.

**Comment by dawidd6** [comment]
Suggestions:
- remove `yaml` input, try to parse as YAML first, fall back to JSON if error occurs
- move the definition into separate file in repository and download this file via API (this is probably for another PR)

**Comment by SMillerDev** [comment]
> remove yaml input, try to parse as YAML first, fall back to JSON if error occurs

JSON is actually valid YAML: https://en.wikipedia.org/wiki/YAML#Comparison_with_JSON

File: label-pull-requests/main.js [hunk 1]
```diff
@@ -10,11 +10,14 @@ async function main() {
...
-        if (core.getBooleanInput("yaml")) {
-            constraints = Object.entries(yaml.load(def)).map(([label, constraint]) => Object.assign({label}, constraint))
-        } else {
+        try {
+            constraints = yaml.load(def)
+        } catch (error) {
             constraints = JSON.parse(def)
         }
+        if (!Array.isArray(constraints)) {
+            constraints = Object.entries(constraints).map(([label, constraint]) => Object.assign({label}, constraint))
```
Comment (dawidd6) [inline label-pull-requests/main.js:19]: What is this for?

Comment (dawidd6) [reply label-pull-requests/main.js:19]: Could we only support arrays for both JSON and YAML please?

**Review by dawidd6 — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @FnControlOption!

**Pull request merged**